docs/source/development.rst

```diff
@@ -4,63 +4,210 @@
-   the status of the test run on the pull request page. There is also a link to
-   Travis so you can inspect the results of the test run, and make additional
-   changes in order to make the tests successful, if needed.
+3. When the pull request is created, the continuous integration tests are run on
+   Travis. You can see the status of the test run on the pull request
+   page. There is also a link to Travis so you can inspect the results of the
+   test run, and make additional changes in order to make the tests successful,
+   if needed. Additionally to the code quality checking tools, unit and
+   regression tests are run as well.
+
+When you create a pull request, assign one or two people to do the review.
+
+
+Code review and merging
+-----------------------
+
+Your pull requests should always be subject to code review. After you create the
+pull request, select one or two contributors and assign them to make a review.
+
+This phase consists of discussion about the introduced changes, suggestions, and
+another requirements made by the reviewers. Anyone who wants to do a review can
+contribute, the reviewer roles are not considered exclusive.
+
+After all of the reviewers' comments have been addressed and the reviewers
+approved the pull request, the pull request can be merged. It is usually a good
+idea to rebase the code to the recent version of master. Assuming your working
+copy is switched to the **master** branch, do:
 
-6. Simultaneously to the test runs, anyone now can look at the code changes by
-   clicking on the "Files changed" tab and comment on individual lines in the
-   diff, and approve or reject the proposed pull request.
+.. code-block:: bash
 
-7. When all the tests are passing and the pull request is approved, it can be
-   merged.
+   $ git pull --rebase
+   $ git checkout your_branch_name
+   $ git rebase master
+
+These commands first update your local copy of master from the remote
+repository, then switch your working copy to the ``your_branch_name`` branch,
+and then rebases the branch on the updated master.
+
+Rebasing is a process in which commits from a branch (``your_branch_name``) are
+applied on a second branch (master), and the new HEAD is marked as the first
+branch.
+
+.. warning:: Rebasing is a process which overwrites history. Therefore be
+             absolutely sure that you know what are you doing. Usually if you
+             work on a branch alone, rebasing is a safe procedure.
+
+When the branch is rebased, you have to force-push it to the repository:
+
+.. code-block:: bash
+
+   $ git push -f origin your_branch_name
+
+This command overwrites the your branch in the remote repository with your local
+branch (which is now rebased on master, and therefore, up-to-date)
+
+.. note:: You can use rebasing also for updating your branch to work with newer
+          versions of master instead of merging the master in the branch. Bear
```

Comment:
Ok, když jsem to psal, tak jsem to po sobě nečet. Jestli ti z toho předchozího přišlo, že update se defaultně dělá rebasem, tak tam tahle note nemusí bejt